cfgen/cfgen

```diff
@@ -1261,6 +1270,31 @@ class ConfPool(ToDhall):
                           f'pvers = {oids_to_dhall(self.pvers)}'])
         return '{ %s }' % args
 
+    @classmethod
+    def gen_sns_tolerance(cls, m0conf: Dict[Oid, Any],
+                          pool_desc: Dict[str, Any]) -> Failures:
+        encls_nr = len(cluster_encls(m0conf))
+        ctrls_per_encl = floor(len(cluster_ctrls(m0conf)) / encls_nr)
+        data_units = pool_desc['data_units']
+        parity_units = pool_desc['parity_units']
+        spare_units = pool_desc.get('spare_units', 0)
+        total_units = data_units + parity_units + spare_units
+        enc_failure_allowed_FP = parity_units / ceil(total_units / encls_nr)
+        enc_failure_allowed = floor(enc_failure_allowed_FP)
+        temp = ctrls_per_encl * enc_failure_allowed
+        ctrl_failure_allowed = min(temp, parity_units)
+        return Failures(0, 0, enc_failure_allowed, ctrl_failure_allowed,
+                        parity_units)
+
+    @classmethod
+    def gen_md_tolerance(cls, m0conf: Dict[Oid, Any],
```

Comment:
gen_md_tolerance should be renamed to gen_dix_tolerance?

Reply:
Dix is also meta data thus md.